cli/azd/internal/repository/infra_confirm.go

```diff
@@ -194,3 +200,26 @@ func (i *Initializer) infraSpecFromDetect(
 
 	return spec, nil
 }
+
+func (i *Initializer) detectPortInDockerfile(
+	filePath string) []int {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return []int{}
+	}
+	defer file.Close()
+
+	var result []int
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "EXPOSE") {
+			var port int
+			_, err := fmt.Sscanf(line, "EXPOSE %d", &port)
+			if err == nil {
+				result = append(result, port)
+			}
+		}
+	}
+	return result
```

Comment:
Instead of us manually reading the dockerfile to discover the exposed port I would suggest we look into consuming existing OSS Go modules that already provide this functionality.

A few to consider are the following:

**moby/buildkit:**
Part of the Moby project (the same project that develops Docker itself), the BuildKit library can be used to parse Dockerfiles. It provides low-level access to the Dockerfile AST (Abstract Syntax Tree) and is widely used in Docker tooling.

**lucor/dockerfile-ast:**
A lighter and simpler AST parser for Dockerfiles with an easy-to-use API to extract metadata from Dockerfiles.

### Example
```go
import (
    "bytes"
    "fmt"
    "github.com/moby/buildkit/frontend/dockerfile/parser"
)

func getExposedPorts(dockerfileContent []byte) ([]string, error) {
    reader := bytes.NewReader(dockerfileContent)
    result, err := parser.Parse(reader)
    if err != nil {
        return nil, err
    }

    var ports []string
    // Loop through the parsed Dockerfile instructions
    for _, child := range result.AST.Children {
        if child.Value == "expose" {
            for _, next := range child.Next {
                ports = append(ports, next.Value) // Get the port value
            }
        }
    }
    return ports, nil
}
```

Reply:
Just offering a different perspective to see how we feel. As the maintainer of `appdetect` package, I think I'd be okay with the current approach, given how lightweight and simple the [EXPOSE](https://docs.docker.com/reference/dockerfile/#expose) syntax is.

The one thing I'd be slightly concerned about introducing AST analysis at the current stage is that `azd` would potentially need to either handle or throw parse errors for `Dockerfile` that it doesn't understand. When these errors occur, it gives a somewhat false impression to the user that `azd` understands Dockerfile at the deeper level than it needs to.

One guiding principle in `appdetect` is that it's fast, and it is minimal -- it will only grep metadata that it understands and ignores the rest. If you have an invalid Dockerfile, that's okay -- we'll grep the file, but not analyze the file for semantic or syntactic correctness. We're not trying to build a Dockerfile, we're just trying to grep to get the few things we needed.

In certain areas in the future, we may go deeper if we think the technology is interesting, but I would recommend staying a little light for Dockerfile for now.